src/app/services/products.service.ts

```diff
@@ -0,0 +1,31 @@
+import { Injectable } from '@angular/core';
+import { CreateProductDto } from '../modules/shared/types/create.product.dto';
+import { HttpClient } from '@angular/common/http';
+import { environment } from 'src/environments/environment';
+import { Observable } from 'rxjs';
+import { ProductDto } from '../modules/shared/types/product.dto';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class ProductService {
+  constructor(private http: HttpClient) {}
+
+  public create(productDto: CreateProductDto): Observable<CreateProductDto> {
+    return this.http.post(
+      `${environment.apiUrl}/products`,
+      productDto
+    ) as Observable<CreateProductDto>;
```

Comment:
Instead of explicitly casting to ```Observable<CreateProductDto>``` you could write something like this 
``` typescript
return this.http.post<CreateProductDto>(
      `${environment.apiUrl}/products`,
      productDto
    )
```